src/Composer/Package/BasePackage.php

```diff
@@ -254,4 +254,23 @@ public static function packageNameToRegexp($allowPattern, $wrap = '{^%s$}i')
 
         return sprintf($wrap, $cleanedAllowPattern);
     }
+
+    /**
+     * Build a regexp from package names, expanding * globs as required
+     *
+     * @param string[] $packageNames
+     * @param string $wrap
+     * @return string
+     */
+    public static function packageNamesToRegexp($packageNames, $wrap = '{^%s$}i')
+    {
+        $packageNames = array_map(
+            static function ($packageName) {
+                return str_replace('\\*', '.*', preg_quote($packageName));
```

Comment:
```suggestion
                return BasePackage::packageNameToRegexp($packageName, '%s');
```

Then you could also make use of this new function in https://github.com/composer/composer/blob/5b16d61762579fcd86507292789f1c92432d9e8b/src/Composer/Repository/FilterRepository.php#L40-L50 as that essentially does the same :)

Reply:
nice, yes this is a good idea! adapted.